src/app/api/community.service.ts

```diff
@@ -0,0 +1,98 @@
+import { Injectable } from '@angular/core';
+import { BaseApiService, Version } from './base-api.service';
+
+export interface Community {
+  name: string;
+  title: string;
+  description: string;
+  sidebar: string;
+  nsfw: boolean;
+  creatorId: number;
+}
+
+export interface CommunityList {
+  communities: Array<string>;
+  totalCount: number;
+  offset: number;
+  size: number;
+  sort: number;
+}
+
+export interface TemplateList {
+  templates: Array<string>;
+  totalCount: number;
+  offset: number;
+  size: number;
+  sort: number;
+}
+
+@Injectable()
+export class CommunityService {
+  // private community: Community;
+  // private name: string;
+
+  constructor(private api: BaseApiService) { }
+
+  // create new community
+  createCommunity(
+    name: string,
+    title: string,
+    description: string,
+    sidebar: string,
+    nsfw: boolean,
+    creatorId: string): Promise<any> {
+    return this.isCommunityNameAvailable(name).then((exists: boolean) => {
+      if (!exists) {
+        this.api.post(Version.v1, 'communities', { name, title, description, sidebar, nsfw, creatorId }).then((community: Community) => {
+        return community;
+        });
+      } else { return 'This community already exists'; }
+    });
+  }
+
+  // get communities from database
+  getCommunities(): Promise<CommunityList> {
+    return this.api.get(Version.v1, 'communities?sort=<top|new>&count=<10>&offset=<0>').then((communitiesList: CommunityList) => {
+      return communitiesList;
+    });
+  }
+
+  // get community details
+  getCommunityDetails(name: string): Promise<Community> {
+    return this.api.get(Version.v1, 'communities/' + name).then((community: Community) => {
```

Comment:
Nitpick: You can use ES6 template literals for var and string concatenation: `communities/${name}`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals